src/test/pybind/test_rbd.py

```diff
@@ -2919,18 +2951,37 @@ def test_group_snap_flags(self):
         eq([], list(self.group.list_snaps()))
 
     def test_group_snap_list_many(self):
+        self.image_names.append(create_image())
+        self.image_names.sort()
+        self.group.add_image(ioctx, self.image_names[0])
+        self.group.add_image(ioctx, self.image_names[1])
+        pool_id = ioctx.get_pool_id()
+
         global snap_name
-        eq([], list(self.group.list_snaps()))
+        assert list(self.group.list_snaps()) == []
         snap_names = []
         for x in range(0, 20):
             snap_names.append(snap_name)
             self.group.create_snap(snap_name)
             snap_name = get_temp_snap_name()
 
         snap_names.sort()
-        answer = [snap['name'] for snap in self.group.list_snaps()]
-        answer.sort()
-        eq(snap_names, answer)
+        gp_snaps_list = self.group.list_snaps()
+        gp_snap_names = []
+        gp_snap_keys = sorted(
+            ['id', 'name', 'state', 'image_snap_name', 'image_snaps'])
+        img_snap_keys = sorted(['pool_id', 'snap_id', 'image_name'])
+        for gp_snap in gp_snaps_list:
+            assert sorted(gp_snap.keys()) == gp_snap_keys
+            gp_snap_names.append(gp_snap['name'])
+            image_names = []
+            for i in gp_snap['image_snaps']:
+                assert sorted(i.keys()) == img_snap_keys
+                assert i['pool_id'] == pool_id
+                image_names.append(i['image_name'])
+            assert sorted(image_names) == self.image_names
+        gp_snap_names.sort()
```

Comment:
How do you choose between `sorted()` (new list) and `sort()` (in-place)?  It doesn't seem to be consistent to me -- is there a (non-obvious) guideline to follow?

Reply:
`sort()` cannot be used for iterables such as dictionary keys. For lists, `sort()` is preferred if it's okay to modify in place instead of creating a new list. I've switched to using `sorted()` throughout.

Reply:
https://stackoverflow.com/a/22442750